openfe/protocols/restraint_utils/geometry/boresch/host.py

```diff
@@ -196,122 +197,136 @@ def find_host_atom_candidates(
+        for ridx in range(len(self.restraints)):
+            # Check angles
+            angle_bounds = True
+            angle_variance = True
+            for i in range(2):
+                bounds = all(
+                    check_angle_not_flat(
+                        angle=angle * unit.radians,
+                        force_constant=self.angle_force_constant,
+                        temperature=self.temperature,
+                    )
+                    for angle in self.results.angles[ridx, i]
                 )
-                for angle in self.results.angles[i]
-            )
-            variance = check_angular_variance(
-                self.results.angles[i] * unit.radians,
-                upper_bound=np.pi * unit.radians,
-                lower_bound=0 * unit.radians,
-                width=1.745 * unit.radians,
-            )
-            angle_bounds &= bounds
-            angle_variance &= variance
-
-        # Check dihedrals
-        dihed_bounds = True
-        dihed_variance = True
-        for i in range(3):
-            bounds = all(
-                check_dihedral_bounds(dihed * unit.radians)
-                for dihed in self.results.dihedrals[i]
-            )
-            variance = check_angular_variance(
-                self.results.dihedrals[i] * unit.radians,
-                upper_bound=np.pi * unit.radians,
-                lower_bound=-np.pi * unit.radians,
-                width=5.23 * unit.radians,
-            )
+                variance = check_angular_variance(
+                    self.results.angles[ridx, i] * unit.radians,
+                    upper_bound=np.pi * unit.radians,
+                    lower_bound=0 * unit.radians,
+                    width=1.745 * unit.radians,
+                )
+                angle_bounds &= bounds
+                angle_variance &= variance
 
-            dihed_bounds &= bounds
-            dihed_variance &= variance
+            # Check dihedrals
+            dihed_bounds = True
+            dihed_variance = True
+            for i in range(3):
+                bounds = all(
+                    check_dihedral_bounds(dihed * unit.radians)
+                    for dihed in self.results.dihedrals[ridx, i]
+                )
+                variance = check_angular_variance(
+                    self.results.dihedrals[ridx, i] * unit.radians,
+                    upper_bound=np.pi * unit.radians,
+                    lower_bound=-np.pi * unit.radians,
+                    width=5.23 * unit.radians,
+                )
 
-        not_collinear = not all(self.results.collinear)
+                dihed_bounds &= bounds
+                dihed_variance &= variance
 
-        self.results.valid = all(
-            [
-                angle_bounds,
-                angle_variance,
-                dihed_bounds,
-                dihed_variance,
-                not_collinear,
-            ]
-        )
+            not_collinear = not all(self.results.collinear)
```

Comment:
```suggestion
            not_collinear = not all(self.results.collinear[ridx])
```
This part is currently failing.

Reply:
🤔 I'm confused as to why the long tests didn't pick this up, will need to look into it.